frontend/src/features/leases/hooks/useLeaseDetail.ts

```diff
@@ -34,14 +31,23 @@ export const useLeaseDetail = (leaseId?: number) => {
         dispatch(hideLoading());
       }
     };
+
     if (leaseId) {
       getLeaseById(leaseId);
     } else {
       toast.error(
         'No valid lease id provided, go back to the lease and license list and select a valid lease.',
       );
     }
-  }, [getLease, leaseId, dispatch]);
+  }, [getLease, dispatch, leaseId, needsUpdate]);
+
+  const refresh = () => {
+    setRefreshIndex(r => r + 1);
```

Comment:
looks a tad hacky. If there is a better way to force an update I would love to hear it!